controller/src/main/java/io/pravega/controller/store/stream/InMemoryStreamMetadataStore.java

```diff
@@ -50,18 +50,21 @@
     @Override
     @Synchronized
     Stream newStream(String scope, String name) {
-        Stream stream = streams.get(scopedStreamName(scope, name));
-        if (stream != null) {
-            return stream;
+        if (streams.containsKey(scopedStreamName(scope, name))) {
+            return streams.get(scopedStreamName(scope, name));
         } else {
-            return new InMemoryStream.NonExistentStream(scope, name);
+            return new InMemoryStream(scope, name);
         }
     }
 
     @Override
     @Synchronized
     Scope newScope(final String scopeName) {
-        return scopes.get(scopeName);
+        if (scopes.containsKey(scopeName)) {
```

Comment:
done.
However, i disagree with your comment here. 
Not protecting reads under lock does not make it 'incorrect'. 
It all depends on the context.. what is the promise of this class.. and does the caller handle stale reads if there is a race with write.
In our case it does. 
The caller is designed to only expect a CAS atomic operation from the store. Thats the contract. 
So any entity that it reads can potentially become stale after having read it.. 
we were protecting writes under lock in this in-memory stream because we want to provide a CAS semantic. So we compare the version and then update. 
Anyhow all that is moot. i have made the changes you have requested.